clients/drcachesim/simulator/cache_simulator.h

```diff
@@ -61,6 +61,7 @@ class cache_simulator_t : public simulator_t
 
     // Exposed to make it easy to test
     bool check_warmed_up();
+    uint64_t knobs_sim_refs() const;
```

Comment:
From the name, I would expect this to return the original option value I passed in, not the *remaining* refs to simulate.   Seems better renamed: s/knobs_/remaining_/ or sthg.